packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx

```diff
@@ -113,13 +114,16 @@ export default function SpanDetail(props: SpanDetailProps) {
               timestamp={traceStartTime}
             />
           )}
-
         <small className="SpanDetail--debugInfo">
-          <Tooltip title="Click ID to add to filter">
-            <span className="SpanDetail--debugLabel" data-label="SpanID:" />{' '}
-            <button className="SpanDetail--debugValue" type="button" onClick={() => addToUiFind(spanID)}>
-              {spanID}
-            </button>
+          <span className="SpanDetail--debugLabel" data-label="SpanID:" /> {spanID}
+          <Tooltip placement="topRight" title="Click to add to filter for deep linking">
+            <Button
+              className="SpanDetail--debugValue ant-btn-icon-only"
+              htmlType="button"
+              onClick={() => addToUiFind(spanID)}
+            >
+              <ZoomIn size="1.4em" />
```

Comment:
I really think it makes sense to treat this as an anchor, which is pretty traditional, instead of appending it to the search field. 

What do you think?

Reply:
Also, the ti icons have a pretty different aesthetic than the others used in the UI. I generally avoid them.

Reply:
Re `ant-btn-icon-only`, I recommend not using Ant's CSS classes, directly, as that will expose the styling to what amounts to being a private API. This will increase the risk of unintended consequences when upgrading.

Reply:
> I really think it makes sense to treat this as an anchor, which is pretty traditional, instead of appending it to the search field.

If the functionality for clicking the icon is to deep link to a `spanID`, not add to the `uiFind`, do you think it'd make sense for this button to instead be "Copy deep link to this span"? On click it'd copy the url but with urlParam for `uiFind` to instead only be the `spanID` instead of whatever is currently present. Replacing the current `uiFind` seems odd, but copying a link to a single id does seem like a useful use case.

Reply:
> Re `ant-btn-icon-only`, I recommend not using Ant's CSS classes, directly, as that will expose the styling to what amounts to being a private API. This will increase the risk of unintended consequences when upgrading.

Generally I agree, but if antd's styling for `antd-btn-icon-only` changes, wouldn't we want this `<Button><SomeIcon /></Button>` to update accordingly? By creating a class that mimic's `antd-btn-icon-only` then on upgrading antd we risk having inconsistent icon buttons. Seems like either way we have a risk when updating the library.

Reply:
#### Which icon(s) to use and append vs replace

> do you think it'd make sense for this button to instead be "Copy deep link to this span"? On click it'd copy the url but with urlParam for `uiFind` to instead only be the `spanID` instead of whatever is currently present.

That sounds like it would work. I would suggest the icon for this be a link icon, maybe the `IoIosLink` in react-icons?

For the case of appending to `uiFind`, the current approach of a magnifying glass with a `+` inside works really well, IMO (not the ti style, but the general idea). Seems like this will be a rare use-case, though. What do you think?

Since these icons will start to look kind of noisy, maybe they can only show up on hover of the span detail panel?

@yurishkuro You had an interest in this functionality, what do you think of having an icon for copying a deep-link to this specific span and one for adding this span to the find (via append)?

#### Using `.antd-btn-icon-only`

One of the main issues with using `.antd-btn-icon-only` (by being renamed, etc) is that it could go away in a future update. In that case, we would miss the updated styling and have broken styling on our `<Button>` that we might miss in inspecting the impact of updating.

By creating a class that mimics the `.antd-btn-icon-only` styling, we would have some risk of inconsistent styling, but I would say that's preferable over risk of broken styling, but still not ideal.

I would think the ideal scenario is to use the React components, as intended. This will generally have the lowest risk of the above scenarios.

Why not go with something like the following?

```jsx
<Button className="AddCustomStyling--etc" ghost size="small">
  <SomeIconHere />
</Button>
```

Reply:
> #### Which icon(s) to use and append vs replace
> > do you think it'd make sense for this button to instead be "Copy deep link to this span"? On click it'd copy the url but with urlParam for `uiFind` to instead only be the `spanID` instead of whatever is currently present.
> 
> That sounds like it would work. I would suggest the icon for this be a link icon, maybe the `IoIosLink` in react-icons?
> 
> For the case of appending to `uiFind`, the current approach of a magnifying glass with a `+` inside works really well, IMO (not the ti style, but the general idea). Seems like this will be a rare use-case, though. What do you think?

I think the append spanID to `uiFind` use-case is really rare, and can be easily done by simply copying and pasting the spanID. I would opt for just one `IoIosLink` icon to copy a deep link url.

> Since these icons will start to look kind of noisy, maybe they can only show up on hover of the span detail panel?

With just one icon that is only visible when the user has expanded the detail row I don't think hiding it until hovering is necessary.

> @yurishkuro You had an interest in this functionality, what do you think of having an icon for copying a deep-link to this specific span and one for adding this span to the find (via append)?

I'm also interested in user's opinions, but as Yuri pointed out on the PR that added `uiFind`, it seems as though adding an already-visible spanID to the `uiFind` doesn't have a purpose outside of deep linking, which would be achieved with the link icon, not the append-to-search icon.

> #### Using `.antd-btn-icon-only`
> One of the main issues with using `.antd-btn-icon-only` (by being renamed, etc) is that it could go away in a future update. In that case, we would miss the updated styling and have broken styling on our `<Button>` that we might miss in inspecting the impact of updating.
> 
> By creating a class that mimics the `.antd-btn-icon-only` styling, we would have some risk of inconsistent styling, but I would say that's preferable over risk of broken styling, but still not ideal.
> 
> I would think the ideal scenario is to use the React components, as intended. This will generally have the lowest risk of the above scenarios.
> 
> Why not go with something like the following?
> 
> ```js-jsx
> <Button className="AddCustomStyling--etc" ghost size="small">
>   <SomeIconHere />
> </Button>
> ```

I hadn't thought that they might remove/rename the class. I also realized that the class isn't intended for `<Button>`s that have children so it could have unintended changes. I added a class to achieve this functionally without using `.antd-btn-icon-only`